src/Servers/Kestrel/Transport.Sockets/src/SocketConnectionOptions.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO.Pipelines;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets
+{
+    /// <summary>
+    /// Options for socket based connections.
+    /// </summary>
+    public class SocketConnectionOptions
+    {
+        /// <summary>
+        /// The <see cref="PipeOptions"/> for socket connections used for input.
+        /// </summary>
+        public PipeOptions InputOptions { get; init; } = new PipeOptions();
+
+        /// <summary>
+        /// The <see cref="PipeOptions"/> for socket connections used for output.
+        /// </summary>
+        public PipeOptions OutputOptions { get; init; } = new PipeOptions();
+
+        /// <summary>
+        /// Delay socket read/write operations until the transport pipe is read from or written to.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to false.
+        /// </remarks>
+        public bool DeferFirstOperation { get; set; }
```

Comment:
Note: this isn't currently used at all

Reply:
Let's just remove it. I think we're going to revert #34458 and close #34639 in favor of this PR.

Reply:
Cool, removed